File: frontend/src/types/tool-log.ts
export interface ToolCallSummary {
  tool_name: string;
  call_count: number;
  success_count: number;
  last_called_at: string | null;
}

export interface ToolCallLogListItem {
  id: string;
  tool_name: string;
  source: string;
  http_method: string | null;
  response_status: number | null;
  latency_ms: number | null;
  success: boolean;
  created_by: string;
  created_at: string;
  conversation_id: string | null;
}

export interface ToolCallLogDetail extends ToolCallLogListItem {
  resource_id: string | null;
  request: Record<string, unknown>;
  response: Record<string, unknown>;
}

File: backend/app/schemas/tool_log.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class ToolCallSummary(BaseModel):
    tool_name: str
    call_count: int
    success_count: int
    last_called_at: datetime | None = None


class ToolCallLogListItem(BaseModel):
    id: UUID
    tool_name: str
    source: str
    http_method: str | None
    response_status: int | None
    latency_ms: int | None
    success: bool
    created_by: str
    created_at: datetime
    conversation_id: UUID | None = None


class ToolCallLogDetail(ToolCallLogListItem):
    resource_id: UUID | None = None
    request: dict[str, Any] = Field(default_factory=dict)
    response: dict[str, Any] = Field(default_factory=dict)
